### DISCUSSION TITLE
Is there a way to set the process workspace while debugging?

### DISCUSSION THREAD

**Discussion by aphage** [NationalSecurityAgency/ghidra#6358] [Debugger]
Is there a way to set the process workspace while debugging?

I don't know how to set the process workspace while debugging.

**Comment by aphage** [comment]
Now I just go into the workspace directory to run ghidra, but I think this is inconvenient.

**Comment by d-millar** [comment]
Am assuming we’re talking about using the debugger vs. debugging Ghidra in, say, Eclipse, yes?   If so, which debugger are you using, i.e. gdb, lldb, dbgeng?

**Comment by d-millar** [comment]
Apologies for the delayed response, by the by - your question somehow escaped our notice!

**Reply by aphage** [reply]
It's dbgeng.

**Reply by aphage** [reply]
On Windows, because I was running ghidra in the ghidra directory, when the debugged program got the current directory, it ended up being the ghidra directory, so I had to go into the debugger directory and re-run ghidra.

**Comment by d-millar** [comment]
@aphage OK,so two answers:

In the old "record"-style version of the debugger (default pre-11.1),  start the debugger from Debugger Targets using "MS dbgeng" or "MS dbgmodel".  Open the "Connectors" node in Objects and select "Launch process", then hit the "Launch (X)" button to the right.  The value of "Dir" gets passed to CreateProcess as the initial directory.

In the newer Python-based traceRMI-style version of the debugger (default for 11.1), that option doesn't exist yet.  Oddly, I was just working on implementing it today.  Will probably be available in 11.2.  

In 11.1, however, both options are available.  You may just have to configure the debugger tool to include "Debugger Targets" and "Objects".
